### PR TITLE
NodeAdmin: new file helper

### PR DESCRIPTION
**Pull request by freva**
* Replaces old `FileHelper` with new that allows for more general filtering.
* Removes node-admin cleanup from `StorageMaintainer`
* Removes metrics from `StorageMaintainer`:
  - `nodes.running_on_host` no longer needed - all node-admins run on host now
  - `nodes.maintenance.fails` no longer needed, will be counted under the general unhandled exceptions
  - `nodes.coredumps` moved to `CleanupTask` in `vespa/hosted`
* `StorageMaintainer` will no longer call to `node-maintainer` for clean up tasks. Only spec verification left.
* Removes throttling on maintenance task.

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    public boolean moveIfExists(Path from, Path to) {
+        try {
+            Files.move(from, to);
+            return true;
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public boolean deleteIfExists(Path path) {
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java]: Make these methods private?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+     */
+    public FileStreamBuilder streamFiles(Path basePath) {
+        return new FileStreamBuilder(basePath, true, false);
+    }
+
+    /**
+     * Creates a {@link FileStreamBuilder} that will by default match all directories and no files
+     * under the given basePath.
+     */
+    public FileStreamBuilder streamDirectories(Path basePath) {
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java]: FileHelper is an unusual class:
 - factory methods like streamDirectories() is non-static
 - builder class is non-static
 - builder class calls method of enclosing class

I'm not entirely convinced this is a good design. For instance the factory method could be static, and the entire body of FileStreamBuilder could be moved to FileHelper, and the FileStreamBuilder inner class can be removed?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java (hunk 1 above)
Comment (freva) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java]: Used elsewhere

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java (hunk 2 above)
Comment (freva) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileHelper.java]: Fixed

**Review by hakonhall — APPROVED** [review]

**Pull request merged**